ClientRuntimes/Python/msrestazure/msrestazure/azure_operation.py

```diff
@@ -136,17 +160,15 @@ def _deserialize(self, response):
          succeeded.
         """
         self.resource = self.get_outputs(response)
-
-        try:
-            if failed(self.resource.provisioning_state):
-                self.status = self.resource.provisioning_state
+        if self.method == 'PUT':
```

Comment:
From my understanding, this line is the only part that is needed to fix #993 - we used to update the status based on the provisioning state for both PUT and PATCH, now only PUT.